src/compile-route.js

```diff
@@ -62,6 +63,55 @@ function areHeadersEqual (currentHeader, expectedHeader) {
     return true;
 }
 
+const pathPattern = /^(https?:\/\/)?([\da-z\.-]+)\.([a-z\.]{2,6})([\/\w \.-]*)*\/?$/;
+const queryPattern = /^(\?)?([0-9A-Za-z]+=[0-9A-Za-z]+(\&)?)+$/;
+
+function getUrlSections (url) {
```

Comment:
can't you use `URL.parse()` here https://nodejs.org/api/url.html

Reply:
Maybe consider url normalization package? e.g. https://github.com/sindresorhus/normalize-url